### ISSUE TITLE
"Table of Contents" Block

### ISSUE THREAD

**Issue by burnuser** [WordPress/gutenberg#11047]
**Is your feature request related to a problem? Please describe.**
In the editor screen of Gutenberg is already a "Table of Contents" (named "Content structure") implemented. It would be nice to have the same feature in the frontend!

**Describe the solution you'd like**
Make already existing "Content structure" available as "Table of Contents" Block.

**Describe alternatives you've considered**
Using a "Table of Contents" Plugin like in pre-Gutenberg times.

**Comment by slimmilkduds** [comment]
There’s a couple of plugins that has a block like this, I use this one myself: https://wordpress.org/plugins/advanced-gutenberg/

**Comment by paaljoachim** [comment]
I agree it would be great to have a "Table of Contents" block or as below issue "Insert Summery" Block.

I believe this issue is similar to what your talking about:
https://github.com/WordPress/gutenberg/issues/7115

**Comment by pratikthink** [comment]
See  #2871, #1760  for more.

**Comment by nicopujol** [comment]
+1 for a table of contents block. Writing long articles now with several H2 and H3's. It would help summarize the article at the top for readers. Thanks

**Comment by Wingo5315** [comment]
There is a block similar to this called [Guidepost](https://sortabrilliant.com/guidepost/) (although I tried using it on my production site and it crashes when I add a new header after adding the block.) 

I had a look at some of the blocks that come with Gutenberg and I never knew that there was a Page Break block. So I think a Table of Contents block would be very useful.

**Comment by mahnunchik** [comment]
+1

**Comment by paaljoachim** [comment]
No news at all. It is one of the many issues that will stand still until a developer picks it up and creates a Pull Request (PR). When a PR is created with some suggestions then design folks will also add some suggestions and it will go a bit forth and back until a design is settled on. Then the developer will implement the design. Get feedback on code. Then when ready get a final code review and the PR becomes merged. We then get the feature into the Gutenberg plugin and later on into WordPress Core.

In the mean time....well those of use who are not developers just sit and wait, and if we really need this feature before this gets into Gutenberg plugin then will will need to find another plugin solution.

**Comment by mahnunchik** [comment]
@talldan @draganescu @ellatrix Could you help? Table of content is really important.

**Comment by SeanDS** [comment]
I'd be happy to try to get [the PR](https://github.com/WordPress/gutenberg/pull/15426) working again. What does rebasing involve? Just checking out the branch then merging in the latest master, then fixing what breaks?

**Comment by paaljoachim** [comment]
I feel we need the advice of a developer here.
@Soean @youknowriad

**Comment by draganescu** [comment]
Howdy @SeanDS, working on that would be super awesome of you. Rebasing usually woks by checking out that branch and running `git rebase master`. However, sometimes when the branch is old rebasing will involve solving multiple conflicts for evey change applied on top of master and it might become tricky.

Therefore a good practice if you are taking over a PR is to create your own from the current master and simply copy the updates from that PR into yours.

**Comment by SeanDS** [comment]
Thanks @draganescu, I copied the changes onto a new branch forked off the current head and made a PR (#21040). Please can people have a look?

**Comment by paaljoachim** [comment]
Please close this PR. As there is another PR open.
@mapk

**Comment by SeanDS** [comment]
> Please close this PR. As there is another PR open.
> @mapk

@paaljoachim this is an issue, not a PR

**Comment by paaljoachim** [comment]
Ops... my mistake.... I was jumping around and noticed a few different PR's and bumped into this issue as well. We will close this issue once the Table of Contents Block is merged.
Thanks Sean..:)

**Comment by mahnunchik** [comment]
It may be helpful, already implemented table of content block https://github.com/maximebj/agb.plugin/tree/master/src/blocks/summary

**Comment by paaljoachim** [comment]
Perhaps @mahnunchik 's above link will help @SeanDS and @ZebulanStanphill

**Comment by ZebulanStanphill** [comment]
@paaljoachim I tried out that plugin, but unfortunately it has the same issue my PR used to have where [trying to use the Undo button wouldn't work](https://github.com/WordPress/gutenberg/pull/21234#issuecomment-605602225). :disappointed:
